image/nitf.ksy

```diff
@@ -0,0 +1,766 @@
+        -orig-id: salvl
+        type: str
+        size: 3
+      - id: graphic_location
+        -orig-id: sloc
+        type: str
+        size: 10
+      - id: first_graphic_bound_loc
+        -orig-id: sbnd1
+        type: str
+        size: 10
+      - id: graphic_color
+        -orig-id: scolor
+        type: str
+        size: 1
+      - id: second_graphic_bound_loc
+        -orig-id: sbnd2
+        type: str
+        size: 10
+      - id: reserved2
+        -orig-id: sres2
+        type: str
+        size: 2
+        doc: 'Reserved'
+      - id: graphics_extended_sub_header
+        type: tre_header
+  tre_header:
+    seq:
+      - id: header_data_length
+        -orig-id: hdl
+        type: str
+        size: 5
+      - id: header_overflow
+        -orig-id: ofl
+        type: str
+        size: 3
+        if: header_data_length.to_i != 0
+      - id: header_data
+        type: u1
+        if: header_data_length.to_i > 2
+        repeat: expr
+        repeat-expr: header_data_length.to_i - 3
+  text_segment:
+    params:
+      - id: idx
+        type: u2
+    seq:
+      - id: text_sub_header
+        size: 1
+      - id: text_data_field
+        size: _parent.header.ltnfo[idx].lt.to_i
+  text_sub_header:
+    seq:
+      - id: txtdt
+        type: str
+        size: 14
+      - id: txtitl
+        type: str
+        size: 80
+      - id: tclasnfo
+        type: clasnfo
+      - id: encryp
+        type: encrypt
+      - id: txtfmt
+        type: str
+        size: 3
+      - id: text_extended_sub_header
+        type: tre_header
+  data_extension_segment:
+    params:
+      - id: idx
+        type: u2
+    seq:
+      - id: data_sub_header
+        type: data_sub_header(_parent.header.ldnfo[idx].ldsh.to_i)
+      - id: data_data_field
+        size: _parent.header.ldnfo[idx].ld.to_i
+  data_sub_header_base:
+    seq:
+      - id: de
+        contents: 'DE'
+      - id: desid
+        type: str
+        size: 25
+      - id: desver
+        type: str
+        size: 2
+      - id: declasnfo
+        type: clasnfo
+  data_sub_header:
+    instances:
+      tre_ofl:
+        value: des_base.desid == 'TRE_OVERFLOW'
+    params:
+      - id: total_size
+        type: u2
+    seq:
+      - id: des_base
+        type: data_sub_header_base
+      - id: overflowed_header_type
+        -orig-id: desoflw
+        type: str
+        size: 6
+        if: tre_ofl
+      - id: data_item_overflowed
+        -orig-id: desitem
+        type: str
+        size: 3
+        if: tre_ofl
+      - id: des_defined_subheader_fields_len
+        -orig-id: desshl
+        type: str
+        size: 4
+      - id: desshf
+        type: str
+        size: des_defined_subheader_fields_len.to_i
+      - id: des_defined_data_field
+        -orig-id: desdata
+        type: str
+        size: >
+          total_size - (2 + 25 + 2 + des_base.declasnfo._sizeof + 
+            (tre_ofl ? 6 + 3 : 0) + 4 + des_defined_subheader_fields_len.to_i)

```

Comment:
Again, the same thing as in https://github.com/kaitai-io/kaitai_struct_formats/pull/294#discussion_r449780897.